chunky/src/java/se/llbit/chunky/renderer/RenderWorkerPool.java

```diff
@@ -149,4 +170,42 @@ public void interrupt() {
       worker.interrupt();
     }
   }
+
+  /**
+   * The future to a job. It may be waited on to finish with {@code awaitFinish}.
+   */
+  public static class RenderJobFuture {
+    private volatile boolean done = false;
+    protected final RenderJob task;
+
+    protected RenderJobFuture(RenderJob task) {
+      this.task = task;
+    }
+
+    protected synchronized void finished() {
+      this.done = true;
+      this.notifyAll();
+    }
+
+    public boolean isDone() {
+      return done;
+    }
+
+    public void awaitFinish() throws InterruptedException {
+      if (done) return;
+      synchronized (this) {
+        while (!done) {
+          this.wait();
+        }
+      }
+    }
+  }
+
+  /**
+   * A render job. This may throw an {@code InterruptedException}.
```

Comment:
That's already in the signature. Use javadoc's `@throws` field to document _when_ it will throw that.